onnxruntime/core/session/provider_registration.cc

```diff
@@ -169,7 +169,7 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider,
   } else {
     ORT_UNUSED_PARAMETER(options);
     status = OrtApis::CreateStatus(ORT_INVALID_ARGUMENT,
-                                   "Unknown provider name. Currently supported values are 'OPENVINO', 'SNPE', 'XNNPACK', 'QNN', 'WEBNN' ,'CoreML', and 'AZURE'");
+                                   "Unknown provider name. Currently supported values are 'DML', 'QNN', 'OpenVINO', 'SNPE', 'XNNPACK', 'WEBNN', 'WebGPU', 'AZURE', 'JS', 'VitisAI', and 'CoreML'");
```

Comment:
Shall we just output the list of GetAvailableProviders().
https://github.com/microsoft/onnxruntime/blob/cda0d14ca981270ed8d76a5bbaffe978d5d05db8/include/onnxruntime/core/session/onnxruntime_cxx_inline.h#L2170